examples/medplum-chart-demo/src/bots/core/charting-utils.ts

```diff
@@ -0,0 +1,374 @@
+      valueQuantity: systolic,
+    });
+  }
+
+  return components;
+}
+
+/**
+ * This function takes condition data and turns it into an array of Condition resources. If the
+ * Condition is being added to the problem list, then an additional Condition resource will be created. For more details see the
+ * Representing Diagnoses docs here: https://www.medplum.com/docs/charting/representing-diagnoses
+ *
+ * @param partialCondition - A partial Condition resource to be added to
+ * @param encounter - The encounter the data is derived from.
+ * @param user - The user creating the Condition resource
+ * @param problemList - A boolean indicating if the condition should be added to the patient's problem list
+ * @returns An array of Condition resources.
+ */
+export function createConditions(
+  partialCondition: Partial<Condition>,
+  encounter: Encounter,
+  user: Practitioner,
+  problemList: boolean
+): Condition[] {
+  const conditions: Condition[] = [];
+
+  // Create a condition for the encounter diagnosis
+  const encounterDiagnosis: Condition = {
+    ...partialCondition,
+    resourceType: 'Condition',
+    subject: encounter.subject as Reference<Patient>,
+    encounter: { reference: getReferenceString(encounter) },
+    recorder: { reference: getReferenceString(user) },
+    asserter: { reference: getReferenceString(user) },
+    category: [
+      {
+        coding: [
+          {
+            system: 'http://hl7.org/fhir/ValueSet/condition-category',
+            code: 'encounter-diagnosis',
+            display: 'Encounter Diagnosis',
+          },
+        ],
+      },
+    ],
+  };
+
+  conditions.push(encounterDiagnosis);
+
+  // If the problem list question was checked, create an additional condition for it
+  if (problemList) {
+    conditions.push({
+      ...partialCondition,
+      resourceType: 'Condition',
+      subject: encounter.subject as Reference<Patient>,
+      encounter: { reference: getReferenceString(encounter) },
+      recorder: { reference: getReferenceString(user) },
+      asserter: { reference: getReferenceString(user) },
+      category: [
+        {
+          coding: [
+            {
+              system: 'http://hl7.org/fhir/ValueSet/condition-category',
+              code: 'problem-list-item',
+              display: 'Problem List Item',
+            },
+          ],
+        },
+      ],
+    });
+  }
+
+  return conditions;
+}
+
+/**
+ * This function takes ClinicalImpression data and creates a ClinicalImpression resource. The ClinicalImpression resource
+ * represents any notes on an encounter in this context.
+ *
+ * @param encounter - The encounter that the data is derived from
+ * @param user - The user creating the ClinicalImpressions
+ * @param note - A string of any notes that a Practitioner may have taken during the encounter
+ * @returns A ClinicalImpression resource
+ */
+export function createClinicalImpressions(
```

Comment:
Since this only returns a single ClinicalImpression, it should be called `createClinicalImpression` (singular)

Alternatively, you could have it return an array with only 0 or 1 elements